### ISSUE TITLE
reproducible Infinite Loop - ListView / IconLabelRenderer combination at UpdateLayout

### ISSUE THREAD

**Issue by Solethia** [jsmarcus/Xamarin.Plugins#18]
**Error located in source**
Edit 4: Added 'Error located in source'

 @jsmarcus I think I've located the actual source of the error. In your [IconLabelRender.cs](https://github.com/jsmarcus/Xamarin.Plugins/blob/2ee8cc7343b7ed4f7395f801ca9848f6cf1fe6e2/Iconize/FormsPlugin.Iconize.Droid/IconLabelRenderer.cs) you hook onto the OnTextChanged Event in the OnElementChanged eventhandler with the followin code.

````
        /// <summary>
        /// Raises the <see cref="E:ElementChanged" /> event.
        /// </summary>
        /// <param name="e">The <see cref="ElementChangedEventArgs{Label}" /> instance containing the event data.</param>
        protected override void OnElementChanged(ElementChangedEventArgs<Label> e)
        {
            base.OnElementChanged(e);

            if (Control == null || Element == null)
                return;

            UpdateText();
            Control.TextChanged += OnTextChanged;
        }

         /// <summary>
        /// Updates the text.
        /// </summary>
        private void UpdateText()
        {
            Control.TextChanged -= OnTextChanged;

            var icon = Plugin.Iconize.Iconize.FindIconForKey(Element.Text);
            if (icon != null)
            {
                Control.Text = $"{icon.Character}";
                Control.Typeface = Plugin.Iconize.Iconize.FindModuleOf(icon).ToTypeface(Context);
            }

            Control.TextChanged += OnTextChanged;
        }

```
You correctly unsubscribe the event `Control.TextChanged -= OnTextChanged;` before you update the text, but afterwards you add it twice, once at the end of your UpdateText() method, and once right after in the OnElementChanged(...) method. This means that next time you do a UpdateText you have one too many event handlers subscribed and it will create a infinite loop, when the text is updated.

I would normally make a pull request, but it seems your build is failing, so I'd rather not mess with it.

**Original Post**
I've encountered what seems like an infinite or near infinite loop, that happens in one of my views with a listview and some IsVisible hide/show logic. It only happens when i have initialized iconize in MainActivity.cs. Removing the inits makes the view run just fine. 

Removing the ListView also stops the problem.

I've reproduced the issue in a selfcontained view.

Edit1: Spelling.
Edit2: Added reproducible page / setup.
Edit3: Fixed code markup.

**MainActivity.cs**
```
using System;
using Android.App;
using Android.Content.PM;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;

namespace EmployeeApp.Droid
{
    [Activity(Label = "EmployeeApp", Icon = "@drawable/icon", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation)]
    public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsApplicationActivity
    {
        private readonly bool EnableIconize = true;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);
            //Android Setup
            ActionBar.SetIcon(Android.Resource.Color.Transparent);
            //Init Plugins
            PluginInit();
            //Init Forms
            FormsInit(bundle);
            try
            {
                LoadApplication(new EmployeeApp.App());
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
                System.Diagnostics.Debugger.Break();
            }
        }

        private void FormsInit(Bundle bundle)
        {
            global::Xamarin.Forms.Forms.Init(this, bundle);

            if (EnableIconize)
                FormsPlugin.Iconize.Droid.IconControls.Init();
        }

        private void PluginInit()
        {
            if (EnableIconize)
                Plugin.Iconize.Iconize.With(new Plugin.Iconize.Fonts.FontAwesomeModule());
        }
    }
}
```
**IconizeInfiniteReproduction.xaml**
```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="EmployeeApp.Content.Components.Stock.IconizeInfiniteReproduction">
  <StackLayout>
    <!--<Switch IsToggled="{Binding ShowSearchResults}"></Switch>-->
    <SearchBar x:Name="searchBar" Placeholder="Find råvare" Text="{Binding SearchText}" TextChanged="SearchTextChanged" IsFocused="{Binding ShowSearchResults}" />
    <ListView IsVisible="{Binding ShowSearchResults}" ItemsSource="{Binding SearchResults}" ItemTapped="SearchSelectIngredient">
      <ListView.ItemTemplate>
        <DataTemplate>
          <ViewCell>
            <Label FontSize="14" Text="{Binding Name}" />
          </ViewCell>
        </DataTemplate>
      </ListView.ItemTemplate>
    </ListView>
  

  <ScrollView IsVisible="{Binding HidePageContent}">
    <StackLayout>
      <Label Text="{Binding SelectedSearch.Name}"></Label>
    </StackLayout>
  </ScrollView>
  </StackLayout>
</ContentPage>
```
**IconizeInfiniteReproduction.xaml.cs**
```
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Xamarin.Forms;

namespace EmployeeApp.Content.Components.Stock
{
	public partial class IconizeInfiniteReproduction : ContentPage
	{
        private IconizeInfiniteReproductionViewModel _Model
        {
            get
            {
                return BindingContext as IconizeInfiniteReproductionViewModel;
            }
            set
            {
                BindingContext = value;
            }
        }

        private List<IconizeInfiniteReproductionSearchViewModel> ServiceDataEmulator { get; set; }
        public IconizeInfiniteReproduction ()
		{
			InitializeComponent();
            ServiceDataEmulator = new List<IconizeInfiniteReproductionSearchViewModel>()
            {
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random0" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random1" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random2" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random3" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random4" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random5" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random6" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random7" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random8" },
                new IconizeInfiniteReproductionSearchViewModel() {Name = "Random9" },
            };
		}

        protected override void OnAppearing()
        {
            base.OnAppearing();
            Initialize();
        }

        private void Initialize()
        {
            _Model = new IconizeInfiniteReproductionViewModel();
            _Model.SelectedSearch = ServiceDataEmulator.First();
        }

        public void SearchTextChanged(object sender, TextChangedEventArgs e)
        {
            var search = e.NewTextValue.ToLower();
            if (string.IsNullOrEmpty(search))
            {
                _Model.SearchResults = new List<IconizeInfiniteReproductionSearchViewModel>();
            }
            else
            {
                var searchPool = ServiceDataEmulator;
                var results = searchPool.Where(x => x.Name.ToLower().Contains(search)).OrderBy(x => x.Name.ToLower().StartsWith(search));
                _Model.SearchResults = results.ToList();
            }
        }

        public void SearchSelectIngredient(object sender, ItemTappedEventArgs e)
        {
            var selected = e.Item as IconizeInfiniteReproductionSearchViewModel;
            if (selected == null)
                return;

            _Model.SelectedSearch = selected;
            _Model.SearchText = string.Empty;
            searchBar.Unfocus();
        }
    }

    public class IconizeInfiniteReproductionViewModel : PropertyChangedBase
    {
        public IconizeInfiniteReproductionViewModel()
        {
            SearchResults = new List<IconizeInfiniteReproductionSearchViewModel>();
        }

        private IconizeInfiniteReproductionSearchViewModel _SelectedSearch;
        public IconizeInfiniteReproductionSearchViewModel SelectedSearch
        {
            get { return _SelectedSearch; }
            set
            {
                SetField(ref _SelectedSearch, value, nameof(SelectedSearch));
            }
        }

        #region [//--- Search ---//]

        private string _SearchText;
        public string SearchText
        {
            get { return _SearchText; }
            set
            {
                SetField(ref _SearchText, value, nameof(SearchText));
            }
        }

        private bool _ShowSearchResults;
        public bool ShowSearchResults
        {
            get { return _ShowSearchResults; }
            set
            {
                SetField(ref _ShowSearchResults, value, nameof(ShowSearchResults));
                OnPropertyChanged(nameof(HidePageContent));
            }
        }

        public bool HidePageContent
        {
            get { return !ShowSearchResults; } // comment to Change
        }

        public List<IconizeInfiniteReproductionSearchViewModel> _SearchResults;
        public List<IconizeInfiniteReproductionSearchViewModel> SearchResults
        {
            get { return _SearchResults; }
            set { SetField(ref _SearchResults, value, nameof(SearchResults)); }
        }
        #endregion [//--- Search END ---//]
    }

    public class IconizeInfiniteReproductionSearchViewModel 
    {
        public string Name { get; set; }
    }

    public class PropertyChangedBase : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;
        protected void OnPropertyChanged(string propertyName)
        {
            //Debug.WriteLine($"PropertyChanged: {propertyName}");
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        protected INotifier SetField<T>(ref T field, T value, string propertyName)
        {
            //Debug.WriteLine($"{this.GetType().Name}#{this.GetHashCode()}.{propertyName} <- {value} (was {field}) - {(EqualityComparer<T>.Default.Equals(field, value) ? "" : "not ")} swallowed");



            field = value;

            OnPropertyChanged(propertyName);
            return new ChangeNotifier(OnPropertyChanged);
        }


        protected INotifier SetProperty<T>(Func<T> getProperty, Action<T> setProperty, T value, string propertyName)
        {


            setProperty(value);

            OnPropertyChanged(propertyName);
            return new ChangeNotifier(OnPropertyChanged);
        }

    }
    public interface INotifier
    {
        INotifier Notify(params string[] propertyNames);

        INotifier OnChange(Action action);
    }
    public class NoChangeNotifier : INotifier
    {
        public INotifier OnChange(Action action)
        {
            return this;
        }

        public INotifier Notify(params string[] propertyNames)
        {
            return this;
        }
    }

    public class ChangeNotifier : INotifier
    {
        private Action<string> _NotifyAction { get; }

        public ChangeNotifier(Action<string> notifyAction)
        {
            _NotifyAction = notifyAction;
        }

        public INotifier Notify(params string[] propertyNames)
        {
            foreach (var propertyName in propertyNames)
                _NotifyAction(propertyName);

            return this;
        }

        public INotifier OnChange(Action action)
        {
            action();
            return this;
        }
    }
}
```

**Stacktrace:**
```
 	[External Code]	
>	0xB in Android.Runtime.JNIEnv.DeleteLocalRef at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.cs:713,4	C#
 	0x3A in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2789,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x46 in FormsPlugin.Iconize.Droid.IconLabelRenderer.UpdateText at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:88,17	C#
 	0x1 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnTextChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:75,13	C#
 	0x1E in Android.Text.TextWatcherImplementor.Android.Text.ITextWatcher.OnTextChanged at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Android.Text/ITextWatcher.cs:91,5	C#
 	0x18 in Android.Text.ITextWatcherInvoker.n_OnTextChanged_Ljava_lang_CharSequence_III at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Text.ITextWatcher.cs:166,4	C#
 	0x2F in Android.Widget.TextView.set_TextFormatted at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2787,6	C#
 	0x15 in Android.Widget.TextView.set_Text at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.TextView.cs:2798,5	C#
 	0x98 in Xamarin.Forms.Platform.Android.LabelRenderer.UpdateText at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Renderers\LabelRenderer.cs:203,5	C#
 	0x64 in Xamarin.Forms.Platform.Android.LabelRenderer.OnElementChanged at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Renderers\LabelRenderer.cs:89,5	C#
 	0x2 in FormsPlugin.Iconize.Droid.IconLabelRenderer.OnElementChanged at D:\Projects\GitHub\Xamarin.Plugins\Iconize\FormsPlugin.Iconize.Droid\IconLabelRenderer.cs:38,13	C#
 	0xFC in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.Label>.SetElement at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:192,4	C#
 	0x2E in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.Label>.Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:128,4	C#
 	0x3C in Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:79,5	C#
 	0x116 in Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:204,6	C#
 	0xD in Xamarin.Forms.Platform.Android.VisualElementPackager.<.ctor>b__6_0 at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:29,50	C#
 	0x2E in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.View>.OnElementChanged at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:282,5	C#
 	0xFC in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.View>.SetElement at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:192,4	C#
 	0x2E in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.View>.Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:128,4	C#
 	0x9A in Xamarin.Forms.Platform.Android.ViewCellRenderer.ViewCellContainer.Update at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Cells\ViewCellRenderer.cs:101,6	C#
 	0x13 in Xamarin.Forms.Platform.Android.ViewCellRenderer.GetCellCore at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Cells\ViewCellRenderer.cs:17,5	C#
 	0x5D in Xamarin.Forms.Platform.Android.CellRenderer.GetCell at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Cells\CellRenderer.cs:46,4	C#
 	0x28 in Xamarin.Forms.Platform.Android.CellFactory.GetCell at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Cells\CellFactory.cs:19,4	C#
 	0x202 in Xamarin.Forms.Platform.Android.ListViewAdapter.GetView at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Renderers\ListViewAdapter.cs:253,4	C#
 	0x1E in Android.Widget.BaseAdapter.n_GetView_ILandroid_view_View_Landroid_view_ViewGroup_ at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Widget.BaseAdapter.cs:443,4	C#
 	0x66 in Android.Views.ViewGroup.Layout at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Views.ViewGroup.cs:3188,5	C#
 	0x63 in Xamarin.Forms.Platform.Android.ViewRenderer<Xamarin.Forms.ListView,Android.Widget.ListView>.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\ViewRenderer.cs:123,4	C#
 	0x8 in Xamarin.Forms.Platform.Android.ListViewRenderer.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Renderers\ListViewRenderer.cs:165,4	C#
 	0x10 in Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:240,4	C#
 	0x16 in Android.Runtime.JNIEnv.CallNonvirtualVoidMethod at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:444,4	C#
 	0xF7 in Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:223,6	C#
 	0xC6 in Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:89,5	C#
 	0xE in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.ListView>.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:137,5	C#
 	0x3F in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.View>.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:310,5	C#
 	0x10 in Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:240,4	C#
 	0x16 in Android.Runtime.JNIEnv.CallNonvirtualVoidMethod at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:444,4	C#
 	0xF7 in Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:223,6	C#
 	0xC6 in Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:89,5	C#
 	0xE in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.View>.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:137,5	C#
 	0x3F in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.Page>.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:310,5	C#
 	0x10 in Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:240,4	C#
 	0x16 in Android.Runtime.JNIEnv.CallNonvirtualVoidMethod at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:444,4	C#
 	0xF7 in Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:223,6	C#
 	0xC6 in Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:89,5	C#
 	0xE in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.Page>.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:137,5	C#
 	0x3F in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.NavigationPage>.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:310,5	C#
 	0x8 in Xamarin.Forms.Platform.Android.NavigationRenderer.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Renderers\NavigationRenderer.cs:103,4	C#
 	0x10 in Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:240,4	C#
 	0x16 in Android.Runtime.JNIEnv.CallNonvirtualVoidMethod at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/src/Runtime/JNIEnv.g.cs:444,4	C#
 	0xF7 in Xamarin.Forms.Platform.Android.FormsViewGroup.MeasureAndLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android.FormsViewGroup\obj\Release\generated\src\Xamarin.Forms.Platform.Android.FormsViewGroup.cs:223,6	C#
 	0xC6 in Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:89,5	C#
 	0xE in Xamarin.Forms.Platform.Android.VisualElementRenderer<Xamarin.Forms.NavigationPage>.UpdateLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:137,5	C#
 	0xA7 in Xamarin.Forms.Platform.Android.Platform.Xamarin.Forms.Platform.Android.IPlatformLayout.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\Platform.cs:1007,5	C#
 	0x20 in Xamarin.Forms.Platform.Android.PlatformRenderer.OnLayout at C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Platform.Android\PlatformRenderer.cs:73,4	C#
 	0x12 in Android.Views.ViewGroup.n_OnLayout_ZIIII at /Users/builder/data/lanes/3340/4e275588/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Views.ViewGroup.cs:3427,4	C#
```

**Comment by Solethia** [comment]
@jsmarcus is there any update on this?

**Comment by jsmarcus** [comment]
Sorry for the delay. I'll get this fixed shortly. Thank you for all the work/info.

**Comment by jsmarcus** [comment]
This was fixed by #29. New build with this will release later today.
